src/replication.c

```diff
@@ -1797,6 +1803,10 @@ void readSyncBulkPayload(connection *conn) {
             /* Delete the tempDb as it's useless now. */
             disklessLoadDiscardTempDb(diskless_load_tempDb);
             serverLog(LL_NOTICE, "MASTER <-> REPLICA sync: Discarded temporary DB");
+
+            moduleFireServerEvent(REDISMODULE_EVENT_REPL_ASYNC,
+                        REDISMODULE_SUBEVENT_REPL_ASYNC_COMPLETED,
+                        NULL);
```

Comment:
maybe we better call this event before the call for `swapMainDbWithTempDb`.
@MeirShpilraien WDYT?

Reply:
I believe yes when the main DB (which is now the temp DB) will be deleted, a module will get the free function called, right? I believe a module would like to know that those free happen on the main DB after it was swapped.

Reply:
And maybe also a done notifcation? Or do we get any other notification that will indicate done?

Reply:
Yes, I think we better have both

Reply:
Should it then be:
1. REDISMODULE_SUBEVENT_REPL_ASYNC_SWAPPING_DB
2. REDISMODULE_SUBEVENT_REPL_ASYNC_COMPLETED
?

Reply:
same as the other discussion: https://github.com/redis/redis/pull/9323#discussion_r698019132
there's probably no value in both START and END events, and if we have just one, better have it before the action IMHO.